amd/src/package_search/components/container.js

```diff
@@ -20,177 +20,29 @@
  */
 
 import * as templates from 'core/templates';
-import * as strings from 'core/str';
-import Notification from 'core/notification';
-import {BaseComponent} from 'core/reactive';
-
-export default class extends BaseComponent {
-    getWatchers() {
-        return [
-            {watch: "general.loading:updated", handler: this.updateStatus},
-            {watch: "all:updated", handler: this.renderAll},
-            {watch: "recentlyused:updated", handler: this.renderRecentlyUsed},
-            {watch: "favourites:updated", handler: this.renderFavourites},
-            {watch: "mine:updated", handler: this.renderMine},
-        ];
-    }
-
-    async create() {
-        this.selectors = {
-            ALL_HEADER: `[data-for="all-header"]`,
-            ALL_CONTENT: `[data-for="all-content"]`,
-            RECENTLY_USED_HEADER: `[data-for="recently-used-header"]`,
-            RECENTLY_USED_CONTENT: `[data-for="recently-used-content"]`,
-            FAVOURITES_HEADER: `[data-for="favourites-header"]`,
-            FAVOURITES_CONTENT: `[data-for="favourites-content"]`,
-            MINE_HEADER: `[data-for="mine-header"]`,
-            MINE_CONTENT: `[data-for="mine-content"]`,
-        };
```

Comment:
Wieso machst du das nicht mehr mit selectors?

Reply:
Über die Schleife ist es m.M.n. etwas übersichtlicher, deshalb habe ich die `selectors` entfernt.

Reply:
Ich denke in diesem Fall ist es ok. Aber generell geht es bei den selectors ja nicht um Übersichtlichkeit, sondern um Flexibilität. Die Idee ist, dass die selectors beim Initialisieren der Komponente aus einem Mustache Template überschrieben werden können. So kann man z.B. in einem Theme das Template überschreiben und andere selectors übergeben:

> By[ ]passing as an init-param, any mustache can customize the default CSS selectors, which means the component is more reusable.

https://moodledev.io/docs/guides/javascript/reactive#initialize-a-component-from-a-mustache-template